src/openms/source/QC/FragmentMassError.cpp

```diff
@@ -108,6 +108,105 @@ namespace OpenMS
+    // GET EXPERIMENTAL SPECTRUM MATCHING TO PEPTIDEIDENTIFICTION
+    //-----------------------------------------------------------------------
+
+    if (!pep_id.metaValueExists("spectrum_reference"))
+    {
+      throw Exception::InvalidParameter(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No spectrum reference annotated at peptide identifiction!");
+    }
+    const MSSpectrum& exp_spectrum = exp[map_to_spectrum.at(pep_id.getMetaValue("spectrum_reference").toString())];
+
+    if (exp_spectrum.getMSLevel() != 2)
+    {
+      throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "The matching spectrum of the mzML is not an MS2 Spectrum.");
+    }
+    Precursor::ActivationMethod act_method;
+    if (exp_spectrum.getPrecursors().empty() || exp_spectrum.getPrecursors()[0].getActivationMethods().empty())
+    {
+      if (print_warning) OPENMS_LOG_WARN << "No MS2 activation method provided. Using CID as fallback to compute fragment mass errors." << std::endl;
+      print_warning = false; // only print it once
+      act_method = Precursor::ActivationMethod::CID;
+    }
+    else
+    {
+      act_method = *exp_spectrum.getPrecursors()[0].getActivationMethods().begin();
+    }
+
+    //---------------------------------------------------------------------
+    // CREATE THEORETICAL SPECTRUM
+    //---------------------------------------------------------------------
+    PeakSpectrum theo_spectrum = getTheoSpec_(act_method, seq, charge);
+
+    //-----------------------------------------------------------------------
+    // COMPARE THEORETICAL AND EXPERIMENTAL SPECTRUM
+    //-----------------------------------------------------------------------
+    if (exp_spectrum.empty() || theo_spectrum.empty())
+    {
+      OPENMS_LOG_WARN << "The spectrum with RT: " + String(exp_spectrum.getRT()) + " is empty." << "\n";
+      return;
+    }
+
+    auto exp_spectrum_filtered(exp_spectrum);
+    window_mower_filter.filterPeakSpectrum(exp_spectrum_filtered);
+
+    // stores ppms for one spectrum
+    DoubleList ppms{};
+    DoubleList dalton{};
+
+    // iterator, finds nearest peak of a target container to a given peak in a reference container
+    if (tolerance_unit == FragmentMassError::ToleranceUnit::DA)
+    {
+      using MIV = MatchedIterator<MSSpectrum, DaTrait, true>;
+      MIV mi(theo_spectrum, exp_spectrum_filtered, tolerance);
+      twoSpecErrors(mi, ppms, dalton, accumulator_ppm, counter_ppm);
+    }
+    else
+    {
+      using MIV = MatchedIterator<MSSpectrum, PpmTrait, true>;
+      MIV mi(theo_spectrum, exp_spectrum_filtered, tolerance);
+      twoSpecErrors(mi, ppms, dalton, accumulator_ppm, counter_ppm);
+    }
+
+    //-----------------------------------------------------------------------
+    // WRITE PPM ERROR IN PEPTIDEHIT
+    //-----------------------------------------------------------------------
+    pep_id.getHits()[0].setMetaValue("fragment_mass_error_ppm", ppms);
```

Comment:
Please define string for these user values in Constants.h around here https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/CONCEPT/Constants.h#L280